### PR TITLE
feat: add pre commit hooks

### PR DESCRIPTION
**Pull request by jiacai2050**
# Which issue does this PR close?

Closes #

# Rationale for this change
 
There are some convention in our coding rules, it's annoying to manually fix them.

So PR add git commit hooks to automatically fix them.

# What changes are included in this PR?

Add pre-commit config, it will
- Ensure validation of toml/yaml files
- Ensure Cargo.toml is sorted, depend on https://github.com/DevinR528/cargo-sort
- Ensure license header

# Are there any user-facing changes?

Developers can install those hooks via
```
pre-commit install
```
This command only need to execute once, it will work when `git commit` next time.

# How does this change test

Manually

### CODE REVIEW

File: scripts/license-header.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
```
Comment (Rachelint) [inline scripts/license-header.txt:1]: How about placing the license like [iox](https://github.com/influxdata/influxdb_iox)?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        args:
+          - --license-filepath
+          - scripts/license-header.txt
+          - --comment-style
+          - //
+  - repo: local
+    hooks:
+      - id: cargo-sort
+        name: Check Cargo.toml is sorted
+        entry: cargo-sort
```
[the remark this answers is not in the document]
Comment (jiacai2050) [reply .pre-commit-config.yaml:21]: Not required, cargo.toml will be passed as args directly to cargo-sort

File: scripts/license-header.txt (hunk 1 above)
Comment (jiacai2050) [reply scripts/license-header.txt:1]: There is already an license file in root directory, but it's different with header inserted in source code, so I create this new file.

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**